Dockerfile

```diff
@@ -3,14 +3,46 @@ FROM rust:slim-bullseye as builder
 
 WORKDIR /usr/src/hms-mqtt-publish
 
-# We only copy the files we need. Otherwise the git folders caused an error in the arm/v7 build .
-COPY src src
-COPY build.rs .
-COPY Cargo.toml .
+# The following builds the rust application in three stages:
+#
+# 1. Build the dependencies
+# 2. Build the protobuf files
+# 3. Build the application
+# 
+# This way, each stage is cached and rebuilding the application is faster.
 
-# Compile the application and install it
+
+# Stage 1: Build the dependencies
+
+COPY ./Cargo.toml ./
+RUN mkdir src && \
+    echo "fn main() {println!(\"hello from dependency build\")}" > src/main.rs && \
+    cargo build --release
+
+
+# Stage 2: Build the protobuf files
```

Comment:
this seems like a superfluous step since this is part of the build and hardly takes a second.

Reply:
Yes, you're right. Wasn't sure about the qemu arm builds where every small thing seems take forever to complete, but I just checked and that part is indeed insignificant

Reply:
great, then let's collapse this into two stages and we can get it merged. Thanks!